File: hotdog.js
class Hotdog {
constructor(toppings = []) {
this.base = {
bun: "soft",
sausage: "beef"
};
this.toppings = toppings;
}

addTopping(topping) {
this.toppings.push(topping);
}

describe() {
return {
...this.base,
toppings: this.toppings,
description: `Hotdog with ${this.toppings.join(", ")}`
};
}
}

function buildHotdog(toppings) {
const h = new Hotdog(toppings);
return h.describe();
}

module.exports = { Hotdog, buildHotdog };

File: hotdog.py
class Hotdog:
def **init**(self, bun="soft", sausage="beef"):
self.bun = bun
self.sausage = sausage
self.toppings = []

```
def add_topping(self, topping):
    self.toppings.append(topping)

def to_dict(self):
    return {
        "bun": self.bun,
        "sausage": self.sausage,
        "toppings": self.toppings
    }
```

if **name** == "**main**":
h = Hotdog()
h.add_topping("ketchup")
print(h.to_dict())
